js/handlers.js

```diff
@@ -138,6 +134,16 @@ weechat.factory('handlers', ['$rootScope', '$log', 'models', 'plugins', 'notific
         // prefix + fullname, which would happen otherwise). Else, use null so that full_name is used
         old.trimmedName = obj.short_name.replace(/^[#&+]/, '') || (obj.short_name ? ' ' : null);
         old.prefix = ['#', '&', '+'].indexOf(obj.short_name.charAt(0)) >= 0 ? obj.short_name.charAt(0) : '';
+
+        // After a buffer openes we get the name change event from relay protocol
+        // Here we check our outgoing commands that openes a buffer and switch
+        // to it if we find the buffer name it the list
+        if (_.contains(models.outgoingQueries, old.shortName)) {
```

Comment:
underscore is making this unnecessarily complicated in this instance:
```js
var position = models.outgoingQueries.indexOf(old.shortName);
if (position >= 0) {
    models.outgoingQueries.splice(position, 1);
    models.setActiveBuffer(old.id);
}
```
less code in plain js! :P